provisioning/transport/mqtt/src/mqtt.ts

```diff
@@ -52,11 +53,12 @@ export class Mqtt extends EventEmitter implements X509ProvisioningTransport, Sym
 
       /* Codes_SRS_NODE_PROVISIONING_MQTT_18_010: [ When waiting for responses, `registrationRequest` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>.] */
       /* Codes_SRS_NODE_PROVISIONING_MQTT_18_024: [ When waiting for responses, `queryOperationStatus` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>.] */
-      let match = topic.match(/^\$dps\/registrations\/res\/(.*)\/\?\$rid=(.*)$/);
+      let match = topic.match(/^\$dps\/registrations\/res\/(.*)\/\?(.*)$/);
+      let queryParameters = queryString.parse(match[2]);
```

Comment:
>queryString.parse(match[2]); [](start = 28, length = 28)

probably need to try/catch this

Reply:
queryString.parse does NOT appear to throw.